checker/jdk/nullness/src/java/lang/reflect/Field.java

```diff
@@ -21,25 +21,27 @@ protected Field() {}
-  @Pure public int getInt(@UnknownInitialization @Nullable Object obj) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  @Pure public long getLong(@UnknownInitialization @Nullable Object obj) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  @Pure public float getFloat(@UnknownInitialization @Nullable Object obj) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  @Pure public double getDouble(@UnknownInitialization @Nullable Object obj) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void set(@UnknownInitialization @Nullable Object obj, @Nullable Object value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setBoolean(@UnknownInitialization @Nullable Object obj, boolean value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setByte(@UnknownInitialization @Nullable Object obj, byte value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setChar(@UnknownInitialization @Nullable Object obj, char value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setShort(@UnknownInitialization @Nullable Object obj, short value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setInt(@UnknownInitialization @Nullable Object obj, int value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setLong(@UnknownInitialization @Nullable Object obj, long value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setFloat(@UnknownInitialization @Nullable Object obj, float value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
-  public void setDouble(@UnknownInitialization @Nullable Object obj, double value) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
+  // These Object formal parameters can be null if the field is static, or can be non-null if the
+  // field is field instance. We don't know which. To prevent a possible NullPointerException,
+  // the Nullness Checker should conservatively issue a warning whenever null is passed.
+  @SideEffectFree public @Nullable Object get(@UnknownInitialization Object obj) throws IllegalArgumentException, IllegalAccessException { throw new RuntimeException("skeleton method"); }
```

Comment:
Please also remove the `@UnknownInitialization`. We also don't want the top initialization qualifier, for the same reasons as why we don't want `@Nullable`.

Reply:
Could you explain this comment?  We don't want `@Nullable` because it might lead to `NullPointerException`.  How does an `@UnknownInitialization` argument lead to an exception?

Reply:
It won't lead to an exception, that is true. But it would allow reading information from an object that is still being initialized.
So then the return type should also be `@UnknownInitialization`, shouldn't it?
It seems easier to normally only allow fully initialized objects and require a suppression if the receiver is not.

Reply:
A `@PolyInitialized` annotation seems appropriate in that case.

Reply:
... but until we do so, we could remove `@UnknownInitialization`.

Reply:
Even if we had a `@PolyInitialized`, I think it would be unsound here.
Any field read from a fully-initialized object needs to be fully-initialized, but a field read from an under-initialization object may not be under initialization. So using `@PolyInitialized` would introduce an unsoundness.